source/server/filter_chain_manager_impl.cc

```diff
@@ -28,14 +32,28 @@ Network::Address::InstanceConstSharedPtr fakeAddress() {
 } // namespace
 
 PerFilterChainFactoryContextImpl::PerFilterChainFactoryContextImpl(
-    Configuration::FactoryContext& parent_context, Init::Manager& init_manager)
-    : parent_context_(parent_context), init_manager_(init_manager) {}
+    Configuration::DrainableFactoryContext& parent_context, Init::Manager& init_manager)
+    : parent_context_(parent_context), init_manager_(init_manager),
+      drain_manager_(parent_context.drainManager().createChildManager(parent_context.dispatcher())),
```

Comment:
Do we need a call to createChildManager here, or could we use parent_context.drainManager() directly?

Reply:
I think we could use the `parent_context.drainManager()` _today_ given the code, but if future code wanted to take advantage of the callback interface, then we'd want it to stay the way it is for correctness.